spec/unit/knife/cookbook_metadata_spec.rb

```diff
@@ -139,12 +139,17 @@
   end
 
   describe "validate_metadata_json" do
-    it "should validate the metadata json" do
+    before(:each) do
+      @metadata_mock = double("metadata")
+    end
+
+    it "should generate the metadata from json" do
+      allow(Chef::Cookbook::Metadata).to receive(:new).and_return(@metadata_mock)
+      expect(@metadata_mock).to receive(:name).with("foobar")
       expect(File).to receive(:exist?).with("#{@cookbook_dir}/foobar/metadata.json").
-                                   and_return(true)
-      expect(IO).to receive(:read).with("#{@cookbook_dir}/foobar/metadata.json").
-                               and_return(@json_data)
-      expect(Chef::Cookbook::Metadata).to receive(:validate_json).with(@json_data)
+                                      and_return(true)
+      expect(IO).to receive(:read).with("#{@cookbook_dir}/foobar/metadata.json").and_return(@json_data)
+      expect(@metadata_mock).to receive(:validate_json).with(@json_data)
```

Comment:
this isn't blowing up because metadata_mock needs to be an "instance_double(Chef::Metadata)"  if you start by just changing the double to an instance_double you'll see you can't mock validate_json because that is a method on the class not the instance.  you mocked it here, so rspec is obliviously happy with it, because its not validating that the underlying object actually has a method to mock.